BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/MemoryMappedFileHandler.cs

```diff
@@ -162,6 +164,17 @@ public virtual void Write(byte[] buffer, int byteCount)
             }
         }
 
+        private static string GenerateFileName()
+        {
+            var dateTimeStamp = DateTime.UtcNow.ToString("yyyyMMdd-HHmmss", CultureInfo.InvariantCulture);
+
+            var currentProcess = Process.GetCurrentProcess();
+            var processFileName = Path.GetFileName(currentProcess.MainModule.FileName);
+            var processId = currentProcess.Id;
+
+            return $"{dateTimeStamp}.{processFileName}.{processId}.log";
```

Comment:
Regarding 2: Testing this locally... I don't see multiple files being created unless the config file is changed during runtime.

Reply:
Regarding 3: please see my changes to the Readme and let me know if that makes sense :)

Reply:
> Regarding 2: Testing this locally... I don't see multiple files being created unless the config file is changed during runtime.

I meant it might give customers an impression that we do partition into multiple files. In fact our code only writes in single file.